src/coreclr/src/jit/lsraarm.cpp

```diff
@@ -491,6 +491,8 @@ int LinearScan::BuildNode(GenTree* tree)
 
         case GT_RETURN:
             srcCount = BuildReturn(tree);
+            killMask = getKillSetForReturn();
+            BuildDefsWithKills(tree, 0, RBM_NONE, killMask);
```

Comment:
I am OK with this fix as-is, but I think another option that might be slightly cleaner would be to replace the above 3 lines with:
```
            buildInternalIntRegisterDefForNode(tree, RBM_PROFILER_ENTER_TRASH);
            srcCount = BuildReturn(tree);
            buildInternalRegisterUses();
```
Then target.h and emit.cpp could, I believe, remain unchanged. This basically says to the register allocator: "I need an extra register and it must be r2", which will cause it to ensure that r2 is free at the return. However, I haven't tested this approach. Note that normally I wouldn't do this separately (outside of an existing `BuildXXX` method), but in this case `BuildReturn()` is shared across platforms and already complex.

Reply:
Yeah, I mentioned trying something like that [over in the issue](https://github.com/dotnet/runtime/issues/37223#issuecomment-666012530) but wasn't sure how to pull it off.

Seems like the cleanest solution would be to inject these calls early and handle them like any other call.

Not sure how to proceed -- if you're ok with this fix then approve? Else I can try the above.

Reply:
I'll approve; not sure it's worth iterating on.